OpenRA.Mods.Common/Traits/Player/PlaceBuilding.cs

```diff
@@ -110,7 +110,7 @@ static void PlayBuildAnim(Actor self, ActorInfo unit)
 					&& x.Actor.Info.Traits.Get<ProductionInfo>().Produces.Intersect(bi.Queue).Any())
 					.ToList();
 			var producer = producers.Where(x => x.Actor.IsPrimaryBuilding()).Concat(producers)
-				.FirstOrDefault();
+				.First();
```

Comment:
Which would make keeping it as FirstOfDefault the correct thing to do then - nice catch and my bad.